src/main/scala/org/apache/spark/sql/pulsar/PulsarMetadataReader.scala

```diff
@@ -478,4 +479,26 @@ private[pulsar] case class PulsarMetadataReader(
         }
     }
   }
+  private def createPulsarClient(
```

Comment:
This method share the same logic with `CachedPulsarClient`, why not using this class instead?

Reply:
Fixed this (now the class is using a `CachedPulsarClient` instead).